pkg/openstack/loadbalancer.go

```diff
@@ -3150,45 +3125,64 @@ func (lbaas *LbaasV2) ensureLoadBalancerDeleted(ctx context.Context, clusterName
 
 			// Delete the floating IP only if it was created dynamically by the controller manager.
 			if fip != nil {
+				klog.InfoS("Matching floating IP", "floatingIP", fip.FloatingIP, "description", fip.Description)
 				matched, err := regexp.Match("Floating IP for Kubernetes external service", []byte(fip.Description))
 				if err != nil {
 					return err
 				}
-				klog.Info(fmt.Sprintf("deleting floating IP %s (%s)", fip.FloatingIP, fip.ID))
+
 				if matched {
+					klog.InfoS("Deleting floating IP for service", "floatingIP", fip.FloatingIP, "service", klog.KObj(service))
 					mc := metrics.NewMetricContext("floating_ip", "delete")
 					err := floatingips.Delete(lbaas.network, fip.ID).ExtractErr()
 					if mc.ObserveRequest(err) != nil {
 						return fmt.Errorf("failed to delete floating IP %s for loadbalancer VIP port %s: %v", fip.FloatingIP, portID, err)
 					}
-					klog.V(2).Infof("Floating IP %s deleted for load balancer %s, Service: %s", fip.FloatingIP, loadbalancer.ID, serviceName)
+					klog.InfoS("Deleted floating IP for service", "floatingIP", fip.FloatingIP, "service", klog.KObj(service))
 				}
 			}
 		}
 	}
 
-	// delete the loadbalancer and all its sub-resources.
-	if lbaas.opts.UseOctavia && lbaas.opts.CascadeDelete {
-		deleteOpts := loadbalancers.DeleteOpts{Cascade: true}
-		mc := metrics.NewMetricContext("loadbalancer", "delete")
-		err := loadbalancers.Delete(lbaas.lb, loadbalancer.ID, deleteOpts).ExtractErr()
-		if mc.ObserveRequest(err) != nil {
-			return fmt.Errorf("failed to delete loadbalancer %s: %v", loadbalancer.ID, err)
-		}
+	// For neutron-lbaas
+	if !lbaas.opts.UseOctavia {
+		return lbaas.ensureLoadBalancerDeletedLegacy(loadbalancer)
+	}
 
-		err = waitLoadbalancerDeleted(lbaas.lb, loadbalancer.ID)
-		if err != nil {
-			return fmt.Errorf("failed to wait for deleted loadbalancer %s: %v", loadbalancer.ID, err)
+	if needDeleteLB && lbaas.opts.CascadeDelete {
+		klog.InfoS("Deleting load balancer", "lbID", loadbalancer.ID, "service", klog.KObj(service))
```

Comment:
@lingxiankong everywhere Im seeing, you kept 2 logs L 3153, L 3157  logged by default (not debug logs). Is it required ? We can add Deleted confirmation logs to default and other as debug logs?

Reply:
Yeah, I did that on purpose. First, the default log level for klog is V(2), we don't need to explicitly use `klog.V(2)`. Second, log the starting and ending point of operations is considered the best practice (at least in openstack) in production and is useful for troubleshooting purpose, e.g. an operation is stuck, without the starting point logging, it's hard to know what the system is doing at that moment.